src/Functions/DateTimeTransforms.h

```diff
@@ -672,32 +672,32 @@ struct ToStartOfHourImpl
 {
     static constexpr auto name = "toStartOfHour";
 
-    static inline UInt32 execute(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
+    static UInt32 execute(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
     {
         return static_cast<UInt32>(time_zone.toStartOfHour(t.whole));
     }
 
-    static inline UInt32 execute(UInt32 t, const DateLUTImpl & time_zone)
+    static UInt32 execute(UInt32 t, const DateLUTImpl & time_zone)
     {
         return time_zone.toStartOfHour(t);
     }
 
-    static inline UInt32 execute(Int32, const DateLUTImpl &)
+    static UInt32 execute(Int32, const DateLUTImpl &)
     {
-        throwDateTimeIsNotSupported(name);
+        throwDateIsNotSupported(name);
```

Comment:
Seems Int32 / Uint64 stands for date, and UInt32/ Int64 ..stands for datetime?

Can we use template to connect field type and `date/datetime` in order not to confuse them?

Reply:
I think so, but that would be a separate PR.